gl_extensions/src/gl_extensions.cpp

```diff
@@ -0,0 +1,1841 @@
+            glVertexArrayVertexBuffers                 = (PFNGLVERTEXARRAYVERTEXBUFFERSPROC)glXGetProcAddress(reinterpret_cast<const GLubyte *>("glVertexArrayVertexBuffers"));
+            glVertexArrayAttribBinding                 = (PFNGLVERTEXARRAYATTRIBBINDINGPROC)glXGetProcAddress(reinterpret_cast<const GLubyte *>("glVertexArrayAttribBinding"));
+            glVertexArrayAttribFormat                  = (PFNGLVERTEXARRAYATTRIBFORMATPROC)glXGetProcAddress(reinterpret_cast<const GLubyte *>("glVertexArrayAttribFormat"));
+            glVertexArrayAttribIFormat                 = (PFNGLVERTEXARRAYATTRIBIFORMATPROC)glXGetProcAddress(reinterpret_cast<const GLubyte *>("glVertexArrayAttribIFormat"));
+            glVertexArrayAttribLFormat                 = (PFNGLVERTEXARRAYATTRIBLFORMATPROC)glXGetProcAddress(reinterpret_cast<const GLubyte *>("glVertexArrayAttribLFormat"));
+            glVertexArrayBindingDivisor                = (PFNGLVERTEXARRAYBINDINGDIVISORPROC)glXGetProcAddress(reinterpret_cast<const GLubyte *>("glVertexArrayBindingDivisor"));
+            glGetVertexArrayiv                         = (PFNGLGETVERTEXARRAYIVPROC)glXGetProcAddress(reinterpret_cast<const GLubyte *>("glGetVertexArrayiv"));
+            glGetVertexArrayIndexediv                  = (PFNGLGETVERTEXARRAYINDEXEDIVPROC)glXGetProcAddress(reinterpret_cast<const GLubyte *>("glGetVertexArrayIndexediv"));
+            glGetVertexArrayIndexed64iv                = (PFNGLGETVERTEXARRAYINDEXED64IVPROC)glXGetProcAddress(reinterpret_cast<const GLubyte *>("glGetVertexArrayIndexed64iv"));
+            // for sampler object
+            glCreateSamplers                           = (PFNGLCREATESAMPLERSPROC)glXGetProcAddress(reinterpret_cast<const GLubyte *>("glCreateSamplers"));
+            // for program pipeline object
+            glCreateProgramPipelines                   = (PFNGLCREATEPROGRAMPIPELINESPROC)glXGetProcAddress(reinterpret_cast<const GLubyte *>("glCreateProgramPipelines"));
+            // for query object
+            glCreateQueries                            = (PFNGLCREATEQUERIESPROC)glXGetProcAddress(reinterpret_cast<const GLubyte *>("glCreateQueries"));
+            glGetQueryBufferObjectiv                   = (PFNGLGETQUERYBUFFEROBJECTIVPROC)glXGetProcAddress(reinterpret_cast<const GLubyte *>("glGetQueryBufferObjectiv"));
+            glGetQueryBufferObjectuiv                  = (PFNGLGETQUERYBUFFEROBJECTUIVPROC)glXGetProcAddress(reinterpret_cast<const GLubyte *>("glGetQueryBufferObjectuiv"));
+            glGetQueryBufferObjecti64v                 = (PFNGLGETQUERYBUFFEROBJECTI64VPROC)glXGetProcAddress(reinterpret_cast<const GLubyte *>("glGetQueryBufferObjecti64v"));
+            glGetQueryBufferObjectui64v                = (PFNGLGETQUERYBUFFEROBJECTUI64VPROC)glXGetProcAddress(reinterpret_cast<const GLubyte *>("glGetQueryBufferObjectui64v"));
+        }
+    }
+#endif
+#endif
+}
+
```

Comment:
Review this two files as a whole. there is a difference in initialization of win32 and linux sides.

Reply:
Well, yeah, there obviously would be differences